src/DataStructures/DataBoxTag.hpp

```diff
@@ -44,26 +45,58 @@ using DataBoxString_t = const char* const;
  * \derivedrequires
  * - type alias `type` of the type this DataBoxTag represents
  * - `static constexpr DataBoxString_t` that is the same as the type name
+ *    and named `label`
+ *
+ * \example
+ * \snippet Test_DataBox.cpp databox_tag_example
+ *
  * \see DataBox DataBoxPrefix DataBoxString_t get_tag_name
  */
 struct DataBoxTag {};
 
 /*!
  * \ingroup DataBoxGroup
- * \brief Functional Tags for the DataBox inherit from this type
+ * \brief Marks an item as being a prefix to another tag
  *
  * \details
- * Used to mark a type as being a DataBoxTag where the label is a prefix to the
- * DataBoxTag that is a member type alias `tag`.
+ * Used to mark a type as being a DataBoxTag where the `label` is a prefix to
+ * the DataBoxTag that is a member type alias `tag`. A prefix tag must contain a
+ * type alias named `type` with the type of the Tag it is a prefix to, as well
+ * as a type alias `tag` that is the type of the Tag that this prefix tag is
+ * a prefix for. A prefix tag must also have a `label` equal to the name of
+ * the struct (tag).
  *
  * \derivedrequires
- * - type alias `tag` of the DataBoxTag that the function operated on
- * \see DataBox DataBoxTag DataBoxString_t get_tag_name
+ * - type alias `tag` of the DataBoxTag that this tag is a prefix to
+ * - type alias `type` that is the type that this DataBoxPrefix holds
+ * - DataBoxString_t `label` that is the prefix to the `tag`
+ *
+ * \example
+ * A DataBoxPrefix tag has the structure:
+ * \snippet Test_DataBox.cpp databox_prefix_tag_example
+ *
+ * The name used to retrieve a prefix tag from the DataBox is:
+ * \snippet Test_DataBox.cpp databox_name_prefix
+ *
+ *
+ * \see DataBox DataBoxTag DataBoxString_t get_tag_name ComputeItemTag
  */
 struct DataBoxPrefix : DataBoxTag {};
 
 /*! \ingroup DataBoxGroup
  * \brief Marks a DataBoxTag as being a compute item that executes a function
+ *
+ * \details
+ * A compute item tag wraps a callable such as a function, lambda, or an object
+ * with a call operator. The callable is a member variable with the name
+ * `function`. The compute item tag must also have a label, same as the DataBox
+ * tag. A type alias `argument_tags` that is a typelist of the tags that will
+ * be retrieved from the DataBox and whose data will be passed to the callable.
+ *
```

Comment:
It'd be worth mentioning that compute items tags wraps something which along with being callable must be stateless.